### PR TITLE
uploading the IDP article

### PR DESCRIPTION
**Pull request by Anita-ihuman**

### CODE REVIEW

File: content/en/blog/comparison-of-internal-developer-platforms/index.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
+---
+title: "Comparison of Internal Developer Platforms"
+description: "A comprehensive comparison of Internal Developer Platform vendors and tools."
+lead: "A comprehensive comparison of Internal Developer Platform vendors and tools."
+tags:
+  - Internal Developer Platforms 
+  - Internal Developer Portals
+  - cloud
+  - Developer experince
```
Comment (eyalb181) [inline content/en/blog/comparison-of-internal-developer-platforms/index.md:9]:
```suggestion
  - devex
```

File: content/en/blog/comparison-of-internal-developer-platforms/index.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
+---
+title: "Comparison of Internal Developer Platforms"
+description: "A comprehensive comparison of Internal Developer Platform vendors and tools."
+lead: "A comprehensive comparison of Internal Developer Platform vendors and tools."
+tags:
+  - Internal Developer Platforms 
+  - Internal Developer Portals
```
Comment (eyalb181) [inline content/en/blog/comparison-of-internal-developer-platforms/index.md:7]:
```suggestion
  - idps
```

File: content/en/blog/comparison-of-internal-developer-platforms/index.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
+---
+title: "Comparison of Internal Developer Platforms"
+description: "A comprehensive comparison of Internal Developer Platform vendors and tools."
+lead: "A comprehensive comparison of Internal Developer Platform vendors and tools."
+tags:
+  - Internal Developer Platforms
```
Comment (eyalb181) [inline content/en/blog/comparison-of-internal-developer-platforms/index.md:6]: Remove

File: content/en/blog/comparison-of-internal-developer-platforms/index.md [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+  - Internal Developer Platforms 
+  - Internal Developer Portals
+  - cloud
+  - Developer experince
+  - platform-engineering 
+date: 2024-03-22T06:00:00+00:00
+lastmod: 2024-03-22T06:00:00+00:00
+draft: false
+weight: 50
+images: ["designer.png"]
```
Comment (eyalb181) [inline content/en/blog/comparison-of-internal-developer-platforms/index.md:15]: Image is not used anywhere in the article

Comment (eyalb181) [reply content/en/blog/comparison-of-internal-developer-platforms/index.md:15]: Also, put a MetalBear in there to make it a bit less generic

**Review by eyalb181 — CHANGES_REQUESTED** [review]

**Review by eyalb181 — APPROVED** [review]

File: content/en/blog/comparison-of-internal-developer-platforms/index.md [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+
+{{<figure src="port.png" alt="comparison of idp platforms " height="100%" width="100%">}}
+
+* **Backstage:** [Backstage](https://backstage.io/) is an open-source IDP that allows developers to manage services, infrastructure, documentation, and more, all within a single platform. It features a software catalog that provides various tooling options, where teams can track ownership and metadata of all software in their organization. It supports integration with several offerings like GitHub and GitLab and allows teams to create new components from existing templates. Backstage also comprises a plugin architecture that enables teams to tailor the portal to their specific needs, increasing its compatibility with other systems. This IDP provides an intuitive user interface for developers but has a steep learning curve and requires a lot of manual configuration with resources to manage. Compared to other platform solutions, Backstage offers minimal infrastructure provisioning, observability and security capabilities with no compliance.
```
Comment (t4lz) [inline content/en/blog/comparison-of-internal-developer-platforms/index.md:66]: > security capabilities with no compliance

I am a bit confused by this part. Maybe it should say something like:

> ... and security capabilities, and no compliance solutions

or

> ... and security capabilities, but no compliance solutions

I don't know IDPs or backstage so I can't say what's right here, just that I don't understand that sentence as it is.

File: content/en/blog/comparison-of-internal-developer-platforms/index.md [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+weight: 50
+images: ["idp-comparison.png"]
+contributors: ["Anita Ihuman"]
+---
+
+DevOps emerged to address the need for better developer productivity and a more efficient development lifecycle. However, DevOps teams often find themselves bogged down by operational tasks like security, scaling, and infrastructure management. The demand for robust developer platforms continues to grow, and with the rise of cloud computing, organizations struggle to keep pace with the complexities and speed involved. This is where Internal Developer Platforms (IDPs) come in – a trendy concept in Platform Engineering that promises to streamline processes and boost developer productivity. 
+
+This article dives into what internal developer platforms are, provides a comprehensive comparison of IDP vendors and tools, and details the features they offer. At the end of this article, your organization should be able to make informed decisions on which platform best optimizes your development workflows.
+
+# What is an Internal Developer Platform?
```
Comment (t4lz) [inline content/en/blog/comparison-of-internal-developer-platforms/index.md:22]:
```suggestion
## What is an Internal Developer Platform?
```

Shouldn't be bigger than the main title.

File: content/en/blog/comparison-of-internal-developer-platforms/index.md [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+|Nullstone |Platform as a Service  | <ul><li>Free plan for Individuals</li><li>Early start up plan at $50/user/m</li><li>Growing startup plan at $100/user/m</li><li>Custom plan available</li></ul>  | <ul><li>Dashboard</li><li>System Visibility</li><li>Automated pipeline</li><li>Data management</li><li>Access Controls/Permissions</li></ul>|
+|Port  |Developer Portal/Service Catalog  |<ul><li>Pro plan is free for a max of 15 users</li><li>Custom Enterprise plan is available/user </li></ul>  | <ul><li>Dashboard</li><li>Built in CI/CD</li><li>System Visibility</li><li>Service Discovery</li><li>API management</li><li>Access Controls/Permissions</li></ul>|
+|Backstage  |Developer Portal/Service Catalog  | <ul><li>Open source</li></ul>| <ul><li>Service cataloging</li><li>System visibility</li><li>Plugins marketplace</li></ul>|
+|Kratix  |Platform Orchestrator | <ul><li>Open source</li></ul> | <ul><li>Automated pipeline</li><li>API management</li><li>Service cataloging</li></ul>|
+
+
+
+
+# Wrapping up
```
Comment (t4lz) [inline content/en/blog/comparison-of-internal-developer-platforms/index.md:99]:
```suggestion
## Wrapping up
```

File: content/en/blog/comparison-of-internal-developer-platforms/index.md [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+
+{{<figure src="nullstone.png" alt="comparison of idp platforms " height="100%" width="100%">}}
+
+* **Kratix:** [Kratix](https://kratix.io/) is an open source platform orchestrator that offers platform engineers the capability to build internal platforms. It harnesses the power of Kubernetes and GitOps to empower platform teams with self-service capabilities in maintaining up-to-date platforms. It offers a marketplace where platform engineers can find and distribute [Promises](https://docs.kratix.io/main/guides/writing-a-promise) created by the Kratix team. It also allows seamless integration with other tools like Backstage, Kubernetes operators, and Terraform. Compared to other solutions, Kratix offers minimal scalability, observability, security and compliance capabilities. Currently, it does not support integration with most developer tools on the market.
+
+{{<figure src="kratix.png" alt="comparison of idp platforms " height="100%" width="100%">}}
+
+## IDP Comparison
+
+|  | Category | Pricing  |Features |
```
Comment (t4lz) [inline content/en/blog/comparison-of-internal-developer-platforms/index.md:84]: I think it's excellent you included a table. Could be cool to split the features into their own columns and then mark yes/no or check-mark/empty or check-mark/X with green/red colors for each platform, like in these examples:
- https://en.wikipedia.org/wiki/Comparison_of_database_administration_tools#General
- https://en.wikipedia.org/wiki/Comparison_of_programming_languages#General_comparison
- [some examples from google images](https://www.google.com/search?sca_esv=d1ba00635c671747&gbv=2&q=feature+comparison+table&uds=AMwkrPutq-TxOnWtSQaB1WSjWHshcUtBdL6pdEQz_5MvGqVyzfTgsaGjyAvzEPMY1Mfx9qEXLJWyyzeXRmXChrAeVYZpmUyfnqQKoCj7XUqgKmZYznOWewP7e-R78LHaSTZkhV5NtwhyNXz84ncIIm85JkZ4-enVWV-3C4U593B8wyk97lAVToViDBq7Hi8eX7gHojJ-lAp_vAlBXULl3_iZG2h07ONDbrPEjmjJvC51demrU4XrvhSKrCDdticJQIa1O-CHjw_AbeFhaDYIzaMmr_nRSMgq7yWkD7ENdqElT695UGXaTZU&udm=2&prmd=ivnbz&sa=X&ved=2ahUKEwi7hOKGmY-FAxUoTKQEHSZoDLkQtKgLegQILRAB&biw=1615&bih=1060&dpr=1#vhid=lahg9eTHZKl0MM&vssid=mosaic)

**Review by t4lz — COMMENTED** [review]
Very nice post. Hadn't read everything yet, but I don't have a lot to say on the contents anyways as I don't know anything about IDPs.

Some headlines need to be fixed, and there are some other optional suggestions.

File: content/en/blog/comparison-of-internal-developer-platforms/index.md [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
...and developers a self-service infrastructure for building IDPs. It has three products designed to reduce cognitive load and drive standardization. The [Humanitec workload specification Score](https://developer.humanitec.com/score/overview/) enables developers to specify the resources their workload requires, and deploy code to it on any tech stack, regardless of its underlying technology. The [Platform Orchestrator](https://developer.humanitec.com/platform-orchestrator/overview/) integrates with your CI/CD pipeline to standardize configurations and workflows, eliminating infrastructure bottlenecks. Finally, the Humanitec Portal acts as the user interface on top of the IDP. Its user-friendly interface and automation tools simplify infrastructure setup, accelerate delivery cycles, and foster better collaboration between the development and operations teams. Most large organizations migrating from traditional or legacy tools to platform centric tools find it useful in modernizing their CI/CD infrastructure. Compared to other platform solutions, Humanitec offers minimal observability and scalability capabilities. Its manual configurations through the CLI can be daunting for developers.
```
Comment (t4lz) [inline content/en/blog/comparison-of-internal-developer-platforms/index.md]: > Humanitec offers minimal observability and scalability

I think adding "only" could make it clearer this is negative

> Humanitec only offers minimal observability and scalability

Also in other places in the text I would suggest adding "only" somewhere before the word "minimal".
Could just be me though.

Comment (Anita-ihuman) [reply content/en/blog/comparison-of-internal-developer-platforms/index.md]: I think adding "only" gives a definite conclusion that there are no other challenges to the platform except these. However, a user with a different use case might encounter another issue that we did not mention.

**Review by aviramha — APPROVED** [review]

**Review by eyalb181 — APPROVED** [review]

**Pull request merged**